multimodal/src/autogluon/multimodal/models/ovd.py

```diff
@@ -0,0 +1,241 @@
+from torch import nn
+
+from ..constants import (
+    AUTOMM,
+    BBOX,
+    COLUMN,
+    COLUMN_FEATURES,
+    FEATURES,
+    IMAGE,
+    IMAGE_META,
+    IMAGE_VALID_NUM,
+    LABEL,
+    LOGIT_SCALE,
+    LOGITS,
+    MASKS,
+    PROMPT,
+    TEXT_TOKEN_IDS,
+    TEXT_VALID_LENGTH,
+)
+from .utils import assign_layer_ids, get_column_features, init_weights
+
+logger = logging.getLogger(__name__)
+
+
+class OVDModel(nn.Module):
+    """
+    Support the OVD models.
+    Now support GroundingDINO: https://github.com/IDEA-Research/GroundingDINO
+    """
+
+    def __init__(
+        self,
+        prefix: str,
+        checkpoint_name: str,
+        pretrained: Optional[bool] = True,
+    ):
+        """
+        Load the pretrained CLIP from huggingface transformers.
+
+        Parameters
+        ----------
+        prefix
+            The model prefix.
+        num_classes
+            The number of classes. 1 for a regression task.
+        pretrained
+            Whether using the pretrained weights. If pretrained=True, download the pretrained model.
+        """
+        super().__init__()
+        logger.debug(f"initializing {checkpoint_name}")
+        self.checkpoint_name = checkpoint_name
+
+        self.config, self.model = self.get_ovd_config_and_model(checkpoint_name=checkpoint_name, pretrained=pretrained)
+
+        self.box_threshold = self.config.box_threshold
+        self.text_threshold = self.config.text_threshold
+
+        self.prefix = prefix
+
+        self.name_to_id = self.get_layer_ids()
+        self.head_layer_names = [n for n, layer_id in self.name_to_id.items() if layer_id == 0]
+
+    @property
+    def image_key(self):
+        return f"{self.prefix}_{IMAGE}"
+
+    @property
+    def image_meta_key(self):
+        return f"{self.prefix}_{IMAGE_META}"
+
+    @property
+    def prompt_key(self):
+        return f"{self.prefix}_{PROMPT}"
+
+    @property
+    def image_valid_num_key(self):
+        return f"{self.prefix}_{IMAGE_VALID_NUM}"
+
+    @property
+    def label_key(self):
+        return f"{self.prefix}_{LABEL}"
+
+    @property
+    def text_column_prefix(self):
+        return f"{self.text_token_ids_key}_{COLUMN}"
+
+    @property
+    def image_column_prefix(self):
+        return f"{self.image_key}_{COLUMN}"
+
+    def forward(
+        self,
+        batch: dict,
+        with_logits: bool = False,
+    ):
+        """
+        Parameters
+        ----------
+        batch
+            A dictionary containing the input mini-batch data.
+            We need to use the keys with the model prefix to index required data.
+
+        Returns
+        -------
+            A dictionary with logits and features.
+        """
+        try:
+            from groundingdino.util.utils import get_phrases_from_posmap
+        except ImportError as e:
+            logger.warning(
+                "Please install groundingdino by: pip install git+https://github.com/IDEA-Research/GroundingDINO.git"
+            )
```

Comment:
We already raise the error in `get_ovd_config_and_model`. Do we still need this warning?

Reply:
Removed.